### PR TITLE
doc: Minor improvements to release-process.md

### PR DESCRIPTION
**Pull request by MitchellCash**
- Update broken links
- Improve command to generate list of authors for a release

---

I also note that it asks to ping **wumpus** on IRC, to assist in generating a list of merged pulls and sort them into categories based on labels. I tried to turn this into a simple one-liner as well (something like ``git log --merges --format="- \`%h\` %s (%an)" v0.16.0..v0.16.1``), but it didn't seem to capture everything I needed.

Would it be worthwhile **wumpus** open-sourcing his code into `contrib/devtools` so there is no single point of failure (even if it can manually be worked around).

### CODE REVIEW

File: doc/release-process.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ Before every major release:
 
 ### First time / New builders
 
-If you're using the automated script (found in [contrib/gitian-build.sh](/contrib/gitian-build.sh)), then at this point you should run it with the "--setup" command. Otherwise ignore this.
+If you're using the automated script (found in [contrib/gitian-build.py](/contrib/gitian-build.py)), then at this point you should run it with the "--setup" command. Otherwise ignore this.
```
Comment (MarcoFalke) [inline doc/release-process.md]: Could you please also fix this in `contrib/README.md` and https://github.com/bitcoin-core/docs/blob/master/gitian-building.md#initial-gitian-setup

Comment (mitchellcash) [reply doc/release-process.md]: @MarcoFalke sure can

Comment (mitchellcash) [reply doc/release-process.md]: @MarcoFalke both now done

**Comment by MarcoFalke** [comment]
utACK dc9ed123e9319906df24d65ce9b05202d561e40f

File: doc/release-process.md [hunk 2]
```diff
@@ -50,15 +50,15 @@ and sort them into categories based on labels)
 
 Generate list of authors:
 
-    git log --format='%aN' "$*" | sort -ui | sed -e 's/^/- /'
+    git log --format='- %aN' v(current version, e.g. 0.16.0)..v(new version, e.g. 0.16.1) | sort -ui
```
Comment (fanquake) [inline doc/release-process.md]: nit: Could pass `--ignore-case` to sort

**Comment by fanquake** [comment]
Quickly tested ACK. One nit above.

The output will match below/the release notes post nit:
```
git log --format='- %aN' v0.16.1..v0.16.2rc2 | sort -ui --ignore-case
- 532479301
- Ben Woosley
- Bernhard M. Wiedemann
- Chun Kuan Lee
- Cory Fields
- fanquake
- Gregory Sanders
- joemphilips
- John Newbery
- Kristaps Kaupe
- lmanners
- Luke Dashjr
- MarcoFalke
- Matt Corallo
- Pieter Wuille
- practicalswift
- Sjors Provoost
- Suhas Daftuar
- Wladimir J. van der Laan
```

File: doc/release-process.md (hunk 2 above)
Comment (mitchellcash) [reply doc/release-process.md]: @fanquake right!

Interestingly I was using the `sort` utility on Ubuntu 16.04 and it strangely ignores case by default, so I didn't spot this issue. However, I just tested this on my Mac using both `sort` and `gsort` and it definitely requires this flag.

I have now updated the commit with the changes.

**Comment by mitchellcash** [comment]
@fanquake nit fixed by passing `-fiu` to `sort`.

File: doc/release-process.md [hunk 3]
```diff
@@ -50,15 +50,15 @@ and sort them into categories based on labels)
 
 Generate list of authors:
 
-    git log --format='%aN' "$*" | sort -ui | sed -e 's/^/- /'
+    git log --format='- %aN' v(current version, e.g. 0.16.0)..v(new version, e.g. 0.16.1) | sort -fiu
```
Comment (laanwj) [inline doc/release-process.md]: FWIW the script that I actually use is:
```bash
#!/bin/bash
# Create authors list for changelog
# Example: list_authors.sh v0.9.0rc2..0.9.0
export LC_ALL=C
git log --use-mailmap --format='%aN' "$*" | sort -uif | sed -e 's/^/- /'
```

Comment (mitchellcash) [reply doc/release-process.md]: Would you rather I turn this into a script, like you posted, within the ‘devtools’ directory? Being essentially a single line command I didn’t think it worth it, but happy to do so if there’s consensus.

**Review by randolf — APPROVED** [review]

File: doc/release-process.md (hunk 3 above)
Comment (mitchellcash) [reply doc/release-process.md]: @fanquake because there is no shared `mailmap` file in the repo is that actually required?

Comment (fanquake) [reply doc/release-process.md]: @mitchellcash Fair enough, as this is only an example anyways, lets just leave it as is.

**Comment by fanquake** [comment]
utACK 95464c7

Using the new command I see:
```
git log --format='- %aN' v0.16.1..v0.16.2rc2 | sort -fiu
- 532479301
- Ben Woosley
- Bernhard M. Wiedemann
- Chun Kuan Lee
- Cory Fields
- fanquake
- Gregory Sanders
- joemphilips
- John Newbery
- Kristaps Kaupe
- lmanners
- Luke Dashjr
- MarcoFalke
- Matt Corallo
- Pieter Wuille
- practicalswift
- Sjors Provoost
- Suhas Daftuar
- Wladimir J. van der Laan
```

**Pull request merged**